Packet++/header/NtpLayer.h

```diff
@@ -0,0 +1,456 @@
+         * @return Value in NTP timestamp format 
+         */
+        uint64_t getTransmitTimestamp() const;
+
+        /**
+         * Set the value of transmit timestamp
+         * @param[in] val Value in NTP timestamp format
+         */
+        void setTransmitTimestamp(uint64_t val);
+
+        /**
+         * Get the value of transmit timestamp
+         * @return Value in seconds from Unix Epoch (1 Jan 1970)
+         */
+        double getTransmitTimestampInSecs() const;
+
+        /**
+         * Set the value of transmit timestamp
+         * @param[in] val Value in seconds from Unix Epoch (1 Jan 1970) 
+         */
+        void setTransmitTimestampInSecs(double val);
+
+        /**
+         * Get the value of key identifier
+         * @return Returns the key identifier if exists, returns 0 on unsupported NTP version or key identifier not found
+         */
+        uint32_t getKeyID() const;
+
+        /**
+         * Get the value of digest. 
+         * @param[out] digest 
+         * @return Digest value as hexadecimal string, empty string on unsupported version
+         */
+        std::string getDigest() const;
+
+        /**
+         * Convert NTP short format to seconds from the Unix Epoch
+         * 
+         * @param[in] val Value in NTP short format
+         * @return Value in seconds from Unix Epoch (1 Jan 1970)
+         */
+        static double convertFromShortFormat(const uint32_t val);
+
+        /**
+         * Convert NTP timestamp format to seconds from the Unix Epoch
+         * 
+         * @param[in] val Value in NTP timestamp format
+         * @return Value in seconds from Unix Epoch (1 Jan 1970)
+         */
+        static double convertFromTimestampFormat(const uint64_t val);
+
+        /**
+         * Convert seconds from the Unix Epoch to NTP short format
+         * 
+         * @param[in] val Value in seconds from Unix Epoch (1 Jan 1970)
+         * @return Value in NTP short format
+         */
+        static uint32_t convertToShortFormat(const double val);
+
+        /**
+         * Convert seconds from the Unix Epoch to NTP timestamp format
+         * 
+         * @param[in] val Value in seconds from Unix Epoch (1 Jan 1970)
+         * @return Value in NTP timestamp format
+         */
+        static uint64_t convertToTimestampFormat(const double val);
+
+        /**
+         * A static method to convert timestamp value to ISO8601 date time format
+         * @param[in] timestamp Value in seconds from the Unix Epoch
+         * @return std::string ISO8601 formatted string
+         */
+        static std::string convertToIsoFormat(const double timestamp);
+
+        /**
+         * A static method to convert timestamp value to ISO8601 date time format
+         * @param[in] timestampInNTPformat Value in NTP timestamp format
+         * @return std::string ISO8601 formatted string
+         */
+        static std::string convertToIsoFormat(const uint64_t timestampInNTPformat);
```

Comment:
I'm thinking that since these methods are not unique to the NTP protocol, maybe we can move them to [`GeneralUtils.h`](https://github.com/seladb/PcapPlusPlus/blob/master/Common%2B%2B/header/GeneralUtils.h) and then provide methods like `getTransmitTimestampAsString()`, `getReceiveTimestampAsString()`, etc. WDYT?